### DISCUSSION TITLE
Invalid findOne() invocation

### DISCUSSION THREAD

**Discussion by Inayat-C** [prisma/prisma#2804] [Q&A] [answered]
Hi guys, 

I'm getting the following error when trying to work with the findOne() method: 

```
PrismaClientValidationError: 
Invalid `prisma.users.findOne()` invocation in
/Users/Inu/Desktop/care-app/backend/api/controllers/authentication/register.js:20:35

{
  where: {
    user_email: 'johndoe@gmail.com'
    ~~~~~~~~~~
  }
}

Unknown arg `user_email` in where.user_email for type UsersWhereUniqueInput. Available args:

type UsersWhereUniqueInput {
  id?: Int
}
```

Here's my code:

```
const user = await prisma.users.findOne({
	where: {
		user_email: 'johndoe@gmail.com',
	},
});
```

And here's my schema:

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Users {
  id              Int      @default(autoincrement()) @id
  user_email      String
  user_password   String
  user_registered DateTime
}
```

I'm almost certain it's a silly mistake from my end but just can't seem to figure out what it could be! 

Oh and i did run `npx prisma generate` after introspecting.

**Comment by Inayat-C** [comment, marked as answer]
.....and just as i suspected, it was a silly error from my part :).

As stated in the docs, findOne() lets you find an item on the database that has a UNIQUE attribute.  I was trying to find an email address record on my table which wasn't set to be unique.

**Comment by ludralph** [comment]
Hi @Inayat-C,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
